src/convertToObject.js

```diff
@@ -5,8 +5,21 @@
  *
  * @return {object}
  */
+
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString
+    .split(';')
+    .map((block) => block.trim())
+    .filter((block) => block.includes(':'))
+    .reduce((cssProperties, block) => {
+      const [key, value] = block
+        .split(':')
+        .map((splitBlock) => splitBlock.trim());
```

Comment:
[CODE KNOWLEDGE] - Don't need to make copy before each operation. Moreover, methods like `filter` or `map` don't mutate original array. In this context, the `split` method is not mutating the original string and there is no need to worry about copying or mutating the original `block` string.